profiling/src/internal/profile.rs

```diff
@@ -133,7 +143,8 @@ impl Profile {
         start_time: SystemTime,
         sample_types: &[api::ValueType],
         period: Option<api::Period>,
-    ) -> Self {
+        string_arena_capacity_hint: usize,
```

Comment:
Here and in the ffi, it may be worth specifying the unit here is in bytes maybe?

Reply:
It gets rounded to a page size. There's mostly no consequence for setting this high, because it's virtual memory so you pay for what you actually use. There are some edges though. like on Linux, OOM kill will be more attracted to your process if you have high virtual memory use. And .NET supports 32-bit processes which only has 2 GiB of virtual memory available per process.

I met with Christophe and Gregory and they are concerned about hitting a max, any max, because that will screw up the accuracy of the upscaling, and they would consider this inaccuracy intolerable in .NET. I knew that the data would incomplete, and that may have been okay. But it will also be inaccurate, and that's not.

Making a more complicated and advanced allocator is a possibility, but then the risk surface goes up too. I'm currently exploring an arena which has fixed-sized chunks, and it allocates new chunks as needed. It uses the system allocator to hold these in a `Vec`, which isn't ideal, but that's far less memory than is represented by the chunks, so it would still greatly reduce the effects of commingling with the system allocator. I could avoid this by using an intrusive linked-list, but this slows performance and makes it harder to provide a global offset for it. Anyway, this is the direction I'm looking at right now.